mono/utils/mono-threads.h

```diff
@@ -168,7 +168,7 @@ enum {
 };
 
 typedef union {
-	int32_t raw;
+	int32_t raw; // FIXME: why 32 bit when the struct is 16, and shouldn't these be unsigned?
```

Comment:
It's 32 bits because our atomic operations work on 32bit ints, not 16.

Reply:
That’s good to know, but why have a bitfield like that then?

Reply:
I'm not sure.  I think it was a choice by the original author of the state machine code (another option would be to use bitmasks and shifts to unpack the value).  It's not incorrect or inconvenient so there's no real motivation to rewrite it.